docs/containerization.md

```diff
@@ -6,61 +6,98 @@ using dedicated GPUs. This guide shows you how to put the `ilab` CLI, all of its
 dependencies, and your GPU into a container for an isolated and easily reproducible
 experience.
 
-## Steps to build an image then run a container
-
-**Containerfile:**
-
-```dockerfile
-FROM nvcr.io/nvidia/cuda:12.3.2-devel-ubi9
-RUN dnf install -y python3.11 && dnf install -y openssh && dnf install -y git && dnf install -y python3-pip && dnf install -y make automake gcc gcc-c++
-RUN ssh-keyscan github.com > ~/.ssh/known_hosts
-WORKDIR /instructlab
-RUN python3.11 -m ensurepip
-RUN dnf install -y gcc
-RUN rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
-RUN dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo && dnf repolist && dnf config-manager --set-enabled cuda-rhel9-x86_64 && dnf config-manager --set-enabled cuda && dnf config-manager --set-enabled epel && dnf update -y
-RUN --mount=type=ssh,id=default python3.11 -m pip install --force-reinstall nvidia-cuda-nvcc-cu12 
-RUN export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64" \
-    && export CUDA_HOME=/usr/local/cuda \
-    && export PATH="/usr/local/cuda/bin:$PATH" \
-    && export XLA_TARGET=cuda120 \
-    && export XLA_FLAGS=--xla_gpu_cuda_data_dir=/usr/local/cuda
-RUN --mount=type=ssh,id=default CMAKE_ARGS="-DLLAMA_CUBLAS=on" python3.11 -m pip install --force-reinstall --no-cache-dir llama-cpp-python 
-RUN --mount=type=ssh,id=default python3.11 -m pip install git+ssh://git@github.com/instructlab/instructlab.git@stable
-CMD ["/bin/bash"]
+## Build the `ilab` container image
+
+We encourage you to read the [Containerfile](../containers/cuda/Containerfile).
+understand the following explanation.
+
+To reduce the size of the final image, we do a multi-stage build with the _devel_
+CUDA image for the build stage and a virtualenv. The base image for the final
+stage is the CUDA _runtime_ image, with no build tools and we copy the `site-packages`
+folder from the first stage.
+
+You will also notice that the entrypoint of the container image is `/opt/app-root/bin/ilab`.
+This allows to call the `ilab` command with opening a new shell. See below how it
+is beneficial when combined with an alias.
+
+For convenience, we have created a `cuda` target in the Makefile, so building is
+as simple as `make cuda`.
+
+The default image name is `localhost/instructlab`, but you can override it with
+the `CONTAINER_PREFIX` environment variable.
+
+## Configure Podman with NVIDIA container runtime
+
+To configure your machine running RHEL 9.4+, you can follow NVIDIA's documentation to
```

Comment:
Why are we pointing to RHEL and not Fedora? This is a community repository.

Reply:
I have removed the reference to RHEL. I also added a comment that the documentation assumes Fedora-like Linux distribution with Podman and SELinux + some hints how to make it work on distros with Docker.